### PR TITLE
Add PortsStatus::on_open

### PR DESCRIPTION
**Pull request by jeanp413**
## Description

In case of forwarding we need to know configuration in advance. Right now it is possible to get to grasp of it only when port is exposed. We deprecated on open action in the exposed state (ExposedPortInfo::on_exposed) and move to PortStatus itself (use PortsStatus::on_open instead).

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

File: components/supervisor-api/status.proto [hunk 1]
```diff
@@ -142,19 +142,27 @@ enum PortVisibility {
...
     notify = 3;
     notify_private = 4;
 }
+enum OnPortOpenAction {
+    _ignore = 0;
+    _open_browser = 1;
+    _open_preview = 2;
+    _notify = 3;
+    _notify_private = 4;
+}
```
Comment (jeanp413) [inline components/supervisor-api/status.proto:159]: Not sure about naming here as I can use the same names in OnPortExposedAction

**Comment by mustard-mh** [comment]
Any context about why we need these changes?

**Comment by jeanp413** [comment]
I sync @akosyakov today and talked about ports in vscode and we need this info even if the port is not exposed so we can update Machine/settings.json

**Comment by akosyakov** [comment]
@mustard-mh I've updated the PR description to clarify for future readers.

**Review by akosyakov — APPROVED** [review]
lgtm

**Pull request merged**